openskistats/changeset.py

```diff
@@ -0,0 +1,84 @@
+import requests
+import xml.etree.ElementTree as ET
+import polars as pl
+from openskistats.analyze import load_runs_pl, load_ski_areas_pl
+
+import re
+
+def parse_osm_url(url: str):
+    """Extract type and ID from an OpenStreetMap URL."""
+    match = re.match(r"https://www\.openstreetmap\.org/(way|relation)/(\d+)", url)
+    if match:
+        osm_type, osm_id = match.groups()
+        return osm_type, int(osm_id)
+    return None, None
+
+def fetch_changeset_history(osm_type: str, osm_id: int):
+    """Fetch the changeset history for a given OSM entity."""
+    url = f"https://api.openstreetmap.org/api/0.6/{osm_type}/{osm_id}/history"
```

Comment:
Looks like our usage is borderline according to their policies https://operations.osmfoundation.org/policies/api/.

If going this route, we should rate limit requests and set up some method of caching. For a complex decision with many ramifications, like the caching method, consider discussion on multiple options, either here or in the issue prior to implementation.

We should identify ourselves as well, similar to

https://github.com/dhimmel/openskistats/blob/3d0e496d8a6b2e8df7e54dd5d666f6ba7596186c/openskistats/openskimap_utils.py#L68-L70

Reply:
Ok, will think about this a bit more, and discuss.